diff --git a/arch/arm64/configs/gki_defconfig b/arch/arm64/configs/gki_defconfig
index 16adb7019..a115509d7 100644
--- a/arch/arm64/configs/gki_defconfig
+++ b/arch/arm64/configs/gki_defconfig
@@ -298,6 +298,8 @@ CONFIG_BT_HCIUART=m
 CONFIG_BT_HCIUART_LL=y
 CONFIG_BT_HCIUART_BCM=y
 CONFIG_BT_HCIUART_QCA=y
+CONFIG_CFG80211=m
+CONFIG_MAC80211=m
 CONFIG_RFKILL=m
 CONFIG_NFC=m
 CONFIG_PCI=y
@@ -379,7 +381,7 @@ CONFIG_USB_NET_AQC111=m
 # CONFIG_WLAN_VENDOR_INTEL is not set
 # CONFIG_WLAN_VENDOR_INTERSIL is not set
 # CONFIG_WLAN_VENDOR_MARVELL is not set
-# CONFIG_WLAN_VENDOR_MEDIATEK is not set
+CONFIG_MT76x2U=m
 # CONFIG_WLAN_VENDOR_RALINK is not set
 # CONFIG_WLAN_VENDOR_REALTEK is not set
 # CONFIG_WLAN_VENDOR_RSI is not set
diff --git a/modules.bzl b/modules.bzl
index 52d92225a..176a4189a 100644
--- a/modules.bzl
+++ b/modules.bzl
@@ -34,6 +34,12 @@ _COMMON_GKI_MODULES_LIST = [
     "drivers/net/usb/r8153_ecm.ko",
     "drivers/net/usb/rtl8150.ko",
     "drivers/net/usb/usbnet.ko",
+    "drivers/net/wireless/mediatek/mt76/mt76.ko",
+    "drivers/net/wireless/mediatek/mt76/mt76-usb.ko",
+    "drivers/net/wireless/mediatek/mt76/mt76x02-lib.ko",
+    "drivers/net/wireless/mediatek/mt76/mt76x02-usb.ko",
+    "drivers/net/wireless/mediatek/mt76/mt76x2/mt76x2-common.ko",
+    "drivers/net/wireless/mediatek/mt76/mt76x2/mt76x2u.ko",
     "drivers/net/wwan/wwan.ko",
     "drivers/usb/class/cdc-acm.ko",
     "drivers/usb/serial/ftdi_sio.ko",
@@ -62,11 +68,13 @@ _COMMON_GKI_MODULES_LIST = [
     "net/ieee802154/ieee802154_socket.ko",
     "net/l2tp/l2tp_core.ko",
     "net/l2tp/l2tp_ppp.ko",
+    "net/mac80211/mac80211.ko",
     "net/mac802154/mac802154.ko",
     "net/nfc/nfc.ko",
     "net/rfkill/rfkill.ko",
     "net/tipc/diag.ko",
     "net/tipc/tipc.ko",
+    "net/wireless/cfg80211.ko",
 ]
 
 # Deprecated - Use `get_gki_modules_list` function instead.
